### ISSUE TITLE
Server-side validation of empty email field yields exception

### ISSUE THREAD

**Issue by NoNoNo** [getgrav/grav#3722]
Test case:

```yaml
---
form:
  client_side_validation: false
  fields:
    email:
      type: email
      validate:
        required: true
---
```

**Steps to reproduce:**

1. Leave e-mail field empty
2. press Enter

**Expected:**

Message about failed validation

**Actual:** 

Runtime error `strpos(): Argument #3 ($offset) must be contained in argument #1 ($haystack)`
in https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Data/Validation.php#L641

```php
if (!(self::typeText($val, $params, $field) && strpos($val, '@', 1))) {
    return false;
}
```

**Comment by mdestafadilah** [comment]
> Test case:
> 
> ```yaml
> ---
> form:
>   client_side_validation: false
>   fields:
>     email:
>       type: email
>       validate:
>         required: true
> ---
> ```
> 
> **Steps to reproduce:**
> 
> 1. Leave e-mail field empty
> 2. press Enter
> 
> **Expected:**
> 
> Message about failed validation
> 
> **Actual:**
> 
> Runtime error `strpos(): Argument #3 ($offset) must be contained in argument #1 ($haystack)` in https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Data/Validation.php#L641
> 
> ```
> if (!(self::typeText($val, $params, $field) && strpos($val, '@', 1))) {
>     return false;
> }
> ```

can you describe your OS dan Grav Version more detailed?

**Comment by NoNoNo** [comment]
Fixed by @rhukster in Commit [b4c6210](https://github.com/getgrav/grav/commit/b4c62101a43051fc7f5349c7d0a5b6085375c1d7) (Grav v1.7.42.2)
